### PR TITLE
Identify and hide reports with solution

### PR DESCRIPTION
**Pull request by patriczek**
This commits are identifying and by default hide all reports and problems which have a solution. 
More information is in commit messages or in issues which describe our problems. 

https://github.com/abrt/faf/issues/502
https://github.com/abrt/faf/issues/503
https://github.com/abrt/faf/issues/504

### CODE REVIEW

File: src/pyfaf/storage/report.py [hunk 1]
```diff
@@ -43,6 +43,7 @@
 
 from pyfaf.utils.storage import format_reason, most_common_crash_function
 from pyfaf.utils.parse import signal2name
+from pyfaf.storage.sf_prefilter import SfPrefilterSolution
```
Comment (jfilak) [inline src/pyfaf/storage/report.py]: A leftover line.

File: src/pyfaf/solutionfinders/__init__.py [hunk 2]
```diff
@@ -136,6 +139,7 @@ def find_solutions_report(report, db=None, finders=None, osr=None):
                     solution = [solution, ]
                 solutions.append((solution_finder.solution_priority, solution))
 
+
```
Comment (jfilak) [inline src/pyfaf/solutionfinders/__init__.py]: Coding style violation.

File: src/pyfaf/solutionfinders/prefilter_solution_finder.py [hunk 3]
```diff
@@ -139,4 +140,14 @@ def find_solution_db_report(self, db, db_report, osr=None):
                         if parser.match(db_frame.symbolsource.path):
                             return solution
 
+        solution = get_sf_prefilter_sol_by_cause(db, cause=db_report.component.name)
```
Comment (jfilak) [inline src/pyfaf/solutionfinders/prefilter_solution_finder.py]: I am lost, why do you add a new function and why do you ignore the return statements above?

File: src/pyfaf/actions/fedmsg_notify.py [hunk 4]
```diff
@@ -65,6 +65,7 @@ def run(self, cmdline, db):
                                   q_yesterday.c.sum_yesterday)
                  .outerjoin(q_today, Report.id == q_today.c.t_report_id)
                  .outerjoin(q_yesterday, Report.id == q_yesterday.c.y_report_id)
+                 .filter(and_(Report.max_certainty.isnot(None), Report.max_certainty != 100))
```
Comment (jfilak) [inline src/pyfaf/actions/fedmsg_notify.py]: Why do you ignore reports without `max_certainty`?

Comment (patriczek) [inline src/pyfaf/actions/fedmsg_notify.py]: Reports without max_certainty doesn't have a solution.

Comment (jfilak) [inline src/pyfaf/actions/fedmsg_notify.py]: Correct! Reports that does not have a solution should be notified through fedmsg, right?

File: src/pyfaf/actions/pull_solution.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            osr = get_report_opsysrelease(db=db, report_id=report.id)
+            solutions = [find_solution(report, db=db, osr=osr)]
+
+            max_certainty = 0
+            for solution in solutions:
+                if solution is not None and solution.certainty > max_certainty:
+                    max_certainty = solution.certainty
+
+            if max_certainty > 0:
+                report.max_certainty = max_certainty
```
Comment (jfilak) [inline src/pyfaf/actions/pull_solution.py]:
```python
if solutions:
    report.max_certainty = max((s.certainty for s in solutions))
```

File: src/pyfaf/storage/migrations/versions/13557f1962e6_ureport_added_certainty.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+down_revision = '89d35a57f82b'
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+
+def upgrade():
+    op.add_column('reports', sa.Column('max_certainty', sa.Integer, nullable=True))
+    op.create_index(op.f('ix_reports_max_certainty'), 'reports', ['max_certainty'], unique=False)
```
Comment (jfilak) [inline src/pyfaf/storage/migrations/versions/13557f1962e6_ureport_added_certainty.py]: What is this index good for?

File: src/pyfaf/solutionfinders/probable_fix_solution_finder.py [hunk 7]
```diff
@@ -42,7 +42,10 @@ def _posr_to_solution(self, posr):
                         url="",
                         note_text=text,
                         note_html=html,
-                        since=since)
+                        since=since,
+                        type=self.name,
+                        certainty=99
```
Comment (jfilak) [inline src/pyfaf/solutionfinders/probable_fix_solution_finder.py]: Why 99? Can you document it somewhere please?

**Comment by jfilak** [comment]
I have on other objections. Does anyone have anything against merging it?

**Comment by sorki** [comment]
Looks good. Thanks.

**Pull request merged**